src/macros.jl

```diff
@@ -368,7 +376,13 @@ function _constraint_macro(args, macro_name::Symbol, parsefun::Function)
     # in a function returning `ConstraintRef`s and give it to `Containers.container`.
     idxvars, indices = Containers._build_ref_sets(_error, c)
 
-    vectorized, parsecode, buildcall = parsefun(_error, x.args...)
+    if x.head == :(:=)
+        vectorized, parsecode, buildcall = parsefun(_error, x.head, x.args...)
+    elseif x.args[1] != :call  || !is_one_argument_constraint(Val(x.args[1]))
+        vectorized, parsecode, buildcall = parsefun(_error, x.args...)
```

Comment:
I tend to prefer having simpler macro code and move the complexity to method dispatch.
Here, you could change
https://github.com/JuliaOpt/JuMP.jl/blob/ed22a474d0285cf3993ecb75e57a70408c5c8316/src/macros.jl#L179-L182
into
```julia
function parse_constraint(_error::Function, sense::Symbol, args...)
    (sense, vectorized) = _check_vectorized(sense)
    vectorized, parse_one_operator_constraint(_error, vectorized, Val(sense), args...)...
end
```

Reply:
This allows to merge two definitions of `parse_constraint`. However, I don't think it's enough: for `:=`, the dispatch must be done on the expression's head.

Reply:
Could you give a macro call that dispatches to the first one with `:=`

Reply:
Sorry, I missed your comment. Here are a few examples: 

```julia
@constraint(m, x := { y <= 4 })
@constraint(m, x := { alldifferent(y, z) })
```

Reply:
Then a simple and easy to extend solution would be to call
```julia
parsefun(_error, Val(x.head), x.args...)
```
and then have a
```julia
parse_constraint(_error::Function, ::Val{:call}, args...) = parse_constraint(_error, args...)
```
Maybe we could have a different name for the `parse_constraint` that takes the `Val` if that simplifies dispatch